security/security-spring/src/main/resources/applicationContext-security.xml

```diff
@@ -680,50 +679,133 @@
 	    <b:bean id="userIdSource" class="org.springframework.social.security.AuthenticationNameUserIdSource"/>
-
-	    <b:bean id="usersConnectionRepository"
+	
+	    <b:bean id="usersConnectionRepository" 
 	            class="org.springframework.social.connect.mem.InMemoryUsersConnectionRepository">
 	        <b:constructor-arg ref="connectionFactoryLocator" />
 	        <b:property name="connectionSignUp" ref="connectionSignUp"/>
 	    </b:bean>
-	    <b:bean id="connectionSignUp" class="org.cbioportal.security.spring.authentication.googleplus.GoogleplusConnectionSignUp"/>
-	    <b:bean id="connectionFactoryLocator" class="org.springframework.social.security.SocialAuthenticationServiceRegistry">
-	        <b:property name="authenticationServices">
-	            <b:list>
-	                <b:bean class="org.springframework.social.google.security.GoogleAuthenticationService">
-	                    <b:constructor-arg value="${googleplus.consumer.key}" />
-	                    <b:constructor-arg value="${googleplus.consumer.secret}" />
-	                    <b:property name="connectionFactory" ref="googleplusConnectionFactory" />
-	                    <!-- Important: The next property name changed from "scope" to "defaultScope" in 1.1.0.M4 -->
-	                    <b:property name="defaultScope" value="email" />
-	                </b:bean>
-	            </b:list>
-	        </b:property>
-	    </b:bean>
-	    <b:bean id="googleplusConnectionFactory" class="org.cbioportal.security.spring.authentication.googleplus.GoogleplusConnectionFactory" >
-	        <b:constructor-arg value="${googleplus.consumer.key}" />
-	        <b:constructor-arg value="${googleplus.consumer.secret}" />
-	    </b:bean>
+	    <b:bean id="connectionSignUp" class="org.cbioportal.security.spring.authentication.social.SocialConnectionSignUp"/>
+	    
+	</b:beans>
+	 
+	 <b:beans profile="social_auth_microsoft">
+		<b:bean id="connectionFactoryLocator"
+			class="org.springframework.social.security.SocialAuthenticationServiceRegistry">
+			<b:property name="authenticationServices">
+				<b:list>
+					<b:bean
+						class="org.springframework.social.live.security.LiveAuthenticationService">
+						<b:constructor-arg
+							value="${microsoftlive.consumer.key}" />
+						<b:constructor-arg
+							value="${microsoftlive.consumer.secret}" />
+						<b:property name="connectionFactory"
+							ref="microsoftliveConnectionFactory" />
+						<b:property name="defaultScope" value="wl.emails" />
+					</b:bean>
+				</b:list>
+			</b:property>
+		</b:bean>
+	
+		<b:bean id="microsoftliveConnectionFactory"
+			class="org.cbioportal.security.spring.authentication.live.LiveConnectionFactory">
+			<b:constructor-arg
+				value="${microsoftlive.consumer.key}" />
+			<b:constructor-arg
+				value="${microsoftlive.consumer.secret}" />
+		</b:bean>
+	</b:beans>
 
-	    <!-- end googleplus beans -->
+	<b:beans profile="social_auth | social_auth_google">
```

Comment:
this type of logic is shown in a few places. I would mention somewhere that `social_auth` means `social_auth_google` for backwards compatibility reasons. Could be in the commit message, the PR description, somewhere in the code. Maybe ideally all those places